src/v/datalake/translation/deps.cc

```diff
@@ -766,14 +766,24 @@ class partition_lag_tracker : public translation_lag_tracker {
         if (highest_translated_offset <= min_offset_for_translation) {
             return _partition->size_bytes();
         }
-        auto log_offset = _partition->log()->to_log_offset(
-          kafka::offset_cast(*highest_translated_offset));
+        const auto highest_translated_log_offset
+          = _partition->log()->to_log_offset(
+            kafka::offset_cast(*highest_translated_offset));
+        const auto max_translatable_offset = _partition->last_stable_offset();
+        /**
+         * It is possible that the last stable offset is not yet established or
+         * simply smaller than the highest translated offset. f.e. during the
+         * partition movement. In such cases, we cannot calculate the backlog.
+         */
+        if (highest_translated_log_offset >= max_translatable_offset) {
```

Comment:
if these are equal shouldn't this return 0? i.e. comparison should be `>` and not `>=`?

Reply:
you are right